src/js/cast/mediamanager.js

```diff
@@ -1055,13 +1055,16 @@ export default function JWMediaManager(receiverManager, container, events, analy
             });
 
             if (hasPreRoll && !mediaStatus.media.duration) {
-            // It is impossible to determine the duration
-            // before playback.
+                // It is impossible to determine the duration
+                // before playback.
                 resolve();
             } else {
-            // Listen for the 'bufferChange' event in order the duration
-            // before playback begins.
-                playerInstance.once('bufferChange', event => {
+                // Listen for the 'meta' event in order the duration
+                // before playback begins.
+                playerInstance.once('meta', event => {
+                    if (!event.duration) {
```

Comment:
Event duration may come later, so we should keep listening until we get it, then turn off the listener.